src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1376,6 +1376,9 @@ private function getServiceCall($id, Reference $reference = null)
             return '$this';
         }
 
+        if ($this->container->hasDefinition($id) && !$this->container->getDefinition($id)->isPublic()) {
+            return "\$this->services[(isset(\$this->services['$id']) || \$this->{$this->generateMethodName($id)}()) && false ?: '$id']";
```

Comment:
@stof comment added
@sstok not possible: the generated code must be version independent so that people can e.g. warump on 7 & deploy on 5 (even if not recommended)